chi/server.py

```diff
@@ -85,211 +89,300 @@ def instance_create_args(
+        print(f"Server: {server.name}")
+        print(f"  ID: {server.id}")
+        print(f"  Status: {server.status}")
+        print(f"  Image Name: {server.image_name}")
+        print(f"  Flavor Name: {server.flavor_name}")
+        print(f"  Network Name: {server.network_name}")
+        print(f"  Addresses: {server.addresses}")
+        print(f"  Created at: {server.created_at}")
+        print(f"  Keypair: {server.keypair.name if server.keypair else 'N/A'}")
+        print(f"  Host ID: {server.host_id}")
+        print(f"  Reservation ID: {server.reservation_id}")
+        print(f"  Host Status: {server.host_status}")
+        print(f"  Hypervisor Hostname: {server.hypervisor_hostname}")
+        print(f"  Is Locked: {server.is_locked}")
+
+    def _show_widget(self, server):
+        html = "<table style='border-collapse: collapse; width: 100%;'>"
+        html += "<tr style='background-color: #f2f2f2;'>"
+        html += "<th style='border: 1px solid #ddd; padding: 8px;'>Attribute</th>"
+        html += f"<th style='border: 1px solid #ddd; padding: 8px;'>{server.name}</th>"
+        html += "</tr>"
+
+        attributes = [
+            'id', 'status', 'image_name', 'flavor_name', 'addresses', 'network_name',
+            'created_at', 'keypair', 'reservation_id', 'host_id', 'host_status',
+            'hypervisor_hostname', 'is_locked'
+        ]
 
+        for attr in attributes:
+            html += "<tr>"
+            html += f"<td style='border: 1px solid #ddd; padding: 8px;'>{attr.replace('_', ' ').title()}</td>"
+            value = getattr(server, attr)
+            if attr == 'addresses':
+                value = self._format_addresses(value)
+            elif attr == 'keypair':
+                value = value.name if value else 'N/A'
+            html += f"<td style='border: 1px solid #ddd; padding: 8px;'>{value}</td>"
+            html += "</tr>"
+
+        html += "</table>"
+        display(HTML(html))
+
+    def _format_addresses(self, addresses):
+        formatted = ""
+        for network, address_list in addresses.items():
+            formatted += f"<strong>{network}:</strong><br>"
+            for address in address_list:
+                formatted += (f"&nbsp;&nbsp;IP: {address['addr']} (v{address['version']})<br>"
+                              f"&nbsp;&nbsp;Type: {address['OS-EXT-IPS:type']}<br>"
+                              f"&nbsp;&nbsp;MAC: {address['OS-EXT-IPS-MAC:mac_addr']}<br>")
+        return formatted
+
+    def associate_floating_ip(self, fip: Optional[str] = None) -> None:
+        associate_floating_ip(self.id, fip)
+
+    def detach_floating_ip(self, fip: str) -> None:
+        detach_floating_ip(self.id, fip)
+
+    def check_connectivity(self, wait: bool = True, port: int = 22, timeout: int = 500,
+                           type: Optional[str] = "widget") -> bool:
+        # Implementation for checking server connectivity
+        pass
+
+    def ssh_connection(self):
```

Comment:
We should just move this ssh method into here: https://github.com/ChameleonCloud/python-chi/blob/master/chi/ssh.py